### ISSUE TITLE
LS should allow numbers after hyphens

### ISSUE THREAD

**Issue by ozra** [gkz/LiveScript#960]
I've been coding LS for yeeears, and finally I think I should issue a complainty complaint... feature wish...
- That I don't have a problem with in any "app-like" code I write, however:
- That I am _very much annoyed with_ in one-off math/calc/analysis code I write (which is a good use-case for LS)

**So after _that_ ado**:
- Number one wish is that hyphen would be allowed with trailing numbers, and that subtraction operator as compensation requires spacing on both sides. 
   - Of course that would be breaking. But!
   - I _bet_ (not _too_ many crowns/dollars/BTC though) that LS coders are so well behaved, blatantly prudent and good, that subtraction is spaced everywhere.
   - As a parenthesis I've done analysis on many code bases of different languages concerning this, in order to promote hyphen as an identifier symbol in certain languages (because of my love for LS of course B-) ), and from that I've found that next to only `for 0 to len-1` (_pseudo syntax to cover for x langs_) was the construct where hyphen was _ever_ used non-spaced. Somewhat good form, is pretty common that is.
   - I love hyphens (fuck, I'm Swedish, when we don't make megacombinedhyphenlackingwords we do add some hyphens to clarify. Maybe that's why. Maybe not. In any way - I'd love to be able to write `foo-bar-3` and `foo-bar-4` etc.

Thoughts?

**Comment by foxbunny** [comment]
I've only been coding LS for a while, on and off, never run into this before, but now that it's brought up, I'd say this is a proper bug.

**Comment by determin1st** [comment]
you mean variable names in code?

**Comment by ozra** [comment]
Haha, yes.

```livescript
a-foo-1 = 47
a-foo-2 = 42
a-foo = 49
a-foo-1 is a-foo - 2
```

**Comment by determin1st** [comment]
You may use alphabet, to follow this style, for example:

a-foo-a
a-foo-b
a-foo-c
..
a-foo-a is a-foo - 2

**Comment by rhendric** [comment]
For that matter, `a-foo1`, `a-foo2`, etc. are also valid identifiers.

I'm of the opinion that this would be too potentially disruptive a change for too little a benefit for LiveScript's current stage of maturity. Seems like a decent thing for any LiveScript successor languages to consider, though.

**Comment by foxbunny** [comment]
Docs say:

> Lisp hackers, you may be pleased to know that you can use dashes in the name of your variables and functions.

In LISP, you can do this:

```lisp
(define foo-2-bar 12)
12
```

In LiveScript you cannot. Therefore, my conclusion is that this is a bug. I don't know why fixing this bug would be disruptive, though. In a language that allows hyphens in variable names, I'd like to believe nobody would even think of writing `foo-12` to mean `foo - 12`.

**Comment by rhendric** [comment]
I regret to inform you that, despite what you'd like to believe, the LiveScript source itself has over 30 occurrences of `i-1`, `op.length-2`, etc. Do you have any reason to believe that the general population of LiveScript programmers is universally more strict in this regard?

**Comment by foxbunny** [comment]
> Do you have any reason to believe that the general population of LiveScript programmers is universally more strict in this regard?

I no longer do. If LiveScript code base itself has such ambiguities, I guess there is no other way than to live with it.

**Comment by gkz** [comment]
This was done on purpose to allow for existing uses where the subtraction operator was used without spaces. This is especially prevalent in array indices, eg. `arr[i-1]`

**Comment by ozra** [comment]
@determin1st - yeah, I know :) But I want `foo-1-mongoose-2`, but... I've survived so far without it. Worse is all languages that don't accept dashes at all :)

@rhendric - seems LiveScripters are less prudent than the typed-langs populace then (that was the majority of code I analyzed to be honest). My "assumption" on the LS-crowd was rather tongue in cheek though, but the facts are welcome ;-). Non-typed-lang users usually code more "hacky" _in my experience_ — so it's not entirely surprising (_I code more "hacky" in LS than in C++, Onyx, Crystal... so it's not to throw blame or anything_)

@gkz - yeah, it's obvious that it's purposefully designed that way, I didn't think it was an error vis a vis specs - I just believe that allowing it would be more useful than non-spaced subtraction :)

Well, I think we can _definitely_ conclude (based on @rhendric's research) that this is a definite no go, I'll close this.
